WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt

```diff
@@ -179,30 +181,100 @@ class PostListViewModel @Inject constructor(
         }
         result
     }
-    val emptyViewState: LiveData<PostListEmptyViewState> by lazy {
-        val result = MediatorLiveData<PostListEmptyViewState>()
+    val emptyViewState: LiveData<PostListEmptyUiState> by lazy {
+        val result = MediatorLiveData<PostListEmptyUiState>()
         val update = {
-            val error = pagedListWrapper.listError.value
-            if (pagedListWrapper.isEmpty.value != false) {
-                if (error != null) {
-                    if (error.type == ListErrorType.PERMISSION_ERROR) {
-                        PostListEmptyViewState.PERMISSION_ERROR
-                    } else PostListEmptyViewState.REFRESH_ERROR
-                } else if (pagedListWrapper.isFetchingFirstPage.value == true) {
-                    PostListEmptyViewState.LOADING
-                } else {
-                    PostListEmptyViewState.EMPTY_LIST
-                }
-            } else {
-                PostListEmptyViewState.HIDDEN_LIST
-            }
+            createEmptyUiState()
         }
         result.addSource(pagedListWrapper.isEmpty) { result.value = update() }
         result.addSource(pagedListWrapper.isFetchingFirstPage) { result.value = update() }
         result.addSource(pagedListWrapper.listError) { result.value = update() }
         result
     }
 
+    private fun createEmptyUiState(): PostListEmptyUiState {
+        return if (pagedListWrapper.isEmpty.value != false) {
+            val error = pagedListWrapper.listError.value
+            val isLoadingFirstPage = pagedListWrapper.isFetchingFirstPage.value == true
+            when {
+                error != null -> createErrorListUiState(error)
+                isLoadingFirstPage -> PostListEmptyUiState.Loading
+                else -> createEmptyListUiState()
+            }
+        } else {
+            PostListEmptyUiState.HiddenList
+        }
+    }
+
+    private fun createErrorListUiState(error: ListError): PostListEmptyUiState {
+        return if (error.type == PERMISSION_ERROR) {
+            PostListEmptyUiState.PermissionsError
+        } else {
+            if (networkUtilsWrapper.isNetworkAvailable()) {
+                PostListEmptyUiState.RefreshError(R.string.error_refresh_posts)
+            } else {
+                PostListEmptyUiState.RefreshError(R.string.no_network_message)
+            }
+        }
+    }
+
+    private fun createEmptyListUiState(): PostListEmptyUiState.EmptyList {
+        return when (postListType) {
+            PUBLISHED -> PostListEmptyUiState.EmptyList(
+                    R.string.posts_published_empty,
+                    R.string.posts_empty_list_button,
+                    this@PostListViewModel::newPost
```

Comment:
Nitpick: I don't think you need the `@PostListViewModel` here, but it doesn't hurt to be explicit either 🤷‍♂️

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9287/commits/814f3bd81f73ce905d99fa165978e697dbffcfcd